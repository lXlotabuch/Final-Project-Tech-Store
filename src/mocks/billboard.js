/* eslint-disable import/prefer-default-export */
export const billboardMock = {
  title: 'gaming monitors',
  img: 'https://fuzzmusic.ru/wp-content/uploads/2020/08/pesni-iz-igry-kiberpank-2077-1.jpg',
  url: '/catalog?categories=gamingMonitors',
  category: 'gaming monitors'
}