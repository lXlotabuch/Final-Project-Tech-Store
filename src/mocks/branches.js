/* eslint-disable import/prefer-default-export */
export const branchesMock = [
  {
    branchName: 'Branch name',
    branchRef: 'kjhkjh3j23kjhkjhk32jhkj4',
  }
]