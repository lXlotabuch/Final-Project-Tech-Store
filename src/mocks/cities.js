/* eslint-disable import/prefer-default-export */
export const citiesMock = [
  {
    CityName: 'Kyiv',
    Ref: '8d5a980d-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Dnipro',
    Ref: 'db5c88f0-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Kharkiv',
    Ref: 'db5c88e0-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Lviv',
    Ref: 'db5c88f5-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Odessa',
    Ref: 'db5c88d0-391c-11dd-90d9-001a92567626'
  },
]

export const cityMock = {
  CityName: 'Kyiv',
  Ref: '8d5a980d-391c-11dd-90d9-001a92567626'
}