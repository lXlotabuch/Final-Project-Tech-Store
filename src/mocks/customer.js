export const mockCustomer = {
  customerNo: '82724063',
  date: '2021-02-19T17:12:50.176+00:00',
  _id: '602ff192ad29b70015add9ab',
  email: 'o.titova.ua@gmail.com',
  isAdmin: false,
  enabled: true,
  firstName: 'someName',
  lastName: 'someLastName',
  login: 'admin',
  password: '$2a$10$i7f6tUO3IT.YsrEGzo/6neTIzoxjhhpWh3Hw3Cme3KLmIAeeyKtFS',
  telephone: '+380731005009',
  gender: 'male',
  ratedProducts: [
    '33823',
    '296354',
    '996598',
    '500227',
    '694186',
    '73109',
    '394414',
    '833228',
    '562546',
    '954601',
    '179158',
    '132036'
  ],
}
export default mockCustomer
