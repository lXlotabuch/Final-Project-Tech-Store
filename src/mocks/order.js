/* eslint-disable import/prefer-default-export */
export const orderMock = {
  deliveryAddress: {country: 'Ukraine', city: 'db5c88f0-391c-11dd-90d9-001a92567626', branch: 'Отделение №2: ул. Академика Янгеля, 40'},
  email: 'o.titova.ua@gmail.com',
  firstName: 'test',
  letterHtml: '<h1>Your order is placed. Our manager will contact you soon.</h1>',
  letterSubject: 'test, thank you for order!',
  mobile: '+380660788828',
  orderNo: '8395086',
  paymentInfo: 'Payment at the time of receipt of the goods',
  shipping: 121,
  totalSum: 117400,
  date: '2021-04-13T14:14:27.478Z',
  _id: '6070d130058f4e00154bbe23'
}