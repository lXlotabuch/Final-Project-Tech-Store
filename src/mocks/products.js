export const productMock = {
  _id: '604ffa554e7ef500153393f9',
  name: 'Monitor 27" AOC C27G2ZU/BK',
  enabled: true,
  currentPrice: 10600,
  previousPrice: 12600,
  categories: 'gamingMonitors',
  imageUrls: [
    'https://i8.rozetka.ua/goods/18842039/aoc_c27g2zu_bk_images_18842039923.jpg',
    'https://i2.rozetka.ua/goods/18842040/aoc_c27g2zu_bk_images_18842040211.jpg',
    'https://i8.rozetka.ua/goods/18842040/aoc_c27g2zu_bk_images_18842040937.jpg',
    'https://i8.rozetka.ua/goods/18842041/aoc_c27g2zu_bk_images_18842041729.jpg'
  ],
  quantity: 44,
  color: 'black',
  productUrl: '/gamingMonitors',
  brand: 'AOC',
  newProduct: 'no',
  reviews: [4, 2, 3, 5, 4, 5, 3, 4, 3, 5, 5, 4, 5],
  params: {
    updateFrequency: '240Hz',
    countryOfmanufacture: 'China',
    displayDiagonal: '27"',
    displayResolution: '1920x1080',
    responseTime: '0,5ms',
    screenSize: '32 inches',
    aspectRatio: '16:9'
  },
  description: 'An immersive 27" super curved gaming monitor with a fast 144Hz refresh rate and 1ms response time, helping you get the most out of your monitor. Features such as AMD FreeSync Premium, QHD resolution, and a curved design make this monitor ultra competitive for any gamer.',
  itemNo: '554194',
}
const productMock2 = {
  ...productMock,
  itemNo: '554195',
  _id: '604ffa554e7ef500153393f0',
}
export const productsMock = [
  {
    _id: '607057d9e448c200152a5047',
    cartQuantity: 1,
    product: productMock
  },
  {
    _id: '607057d9e448c200152a5048',
    cartQuantity: 1,
    product: productMock2
  }
];