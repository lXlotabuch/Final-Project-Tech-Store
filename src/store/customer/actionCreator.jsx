import {
  CLEAR_CUSTOMER,
  SET_CUSTOMER_INFO, SET_ORDERS, START_LOADING, STOP_LOADING
} from './actionType';

export const setCustomerInfo = (payload) => ({
  type: SET_CUSTOMER_INFO,
  payload
})

export const setOrders = (payload) => ({
  type: SET_ORDERS,
  payload
})

export const startLoading = () => ({
  type: START_LOADING
})

export const stopLoading = () => ({
  type: STOP_LOADING
})
export const clearCustomer = () => ({
  type: CLEAR_CUSTOMER
})