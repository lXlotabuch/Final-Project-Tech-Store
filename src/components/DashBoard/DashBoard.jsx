import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Container } from '../common/Container'
import Subscribe from './Subscribed/Subscribe';
import {
  AccountInfo, MyDash, WrapperBlocks, RowBlocks,
} from './StyledDashBoard';
import PasswordInfoChange from './PasswordInfoChange/PasswordInfoChange';
import StyledSpinner from '../StyledSpinner/StyledSpinner'
import Orders from './Orders/Orders';
import { selectCustomerInfo, selectIsLoading } from '../../store/customer/reducer'

const mapStateToProps = (state) => ({
  customerInfo: selectCustomerInfo(state),
  isLoading: selectIsLoading(state)
})

const DashBoard = connect(mapStateToProps, null)(({
  customerInfo,
  isLoading
}) => {
  useEffect(() => {
    window.scrollTo(0, 0);
  }, [])

  if (isLoading) return <StyledSpinner size="large" tip="Loading..." />
  return (
    <Container>
      <MyDash>
        <p>My Dashboard</p>
      </MyDash>
      <AccountInfo>
        <h3>Account Information</h3>
      </AccountInfo>
      <WrapperBlocks>
        <RowBlocks>
          <PasswordInfoChange info={customerInfo} />
        </RowBlocks>
        <RowBlocks>
          <Subscribe
            email={customerInfo.email}
          />
        </RowBlocks>
        <RowBlocks>
          <Orders />
        </RowBlocks>
      </WrapperBlocks>
    </Container>
  );
})
export default DashBoard;
