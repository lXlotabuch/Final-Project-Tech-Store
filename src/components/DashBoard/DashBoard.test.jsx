import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { shallow } from 'enzyme';
import { HashRouter as Router } from 'react-router-dom';
import { machMedia } from '../../mocks/matchMedia.mock';
import DashBoard from './DashBoard'
import { mockCustomer } from '../../mocks/customer';
import { store } from '../../store/index'

describe('Render DashBoard component', () => {
  beforeEach(() => {
    machMedia()
  })

  test('Check if animation renders', () => {
    const results = shallow(
      <Provider store={store}>
        <Router>
          <DashBoard />
        </Router>
      </Provider>,
      false
    )
    expect(results.find('svg')).toBeDefined()
  })
  test('shaphot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <DashBoard
          customerInfo={mockCustomer}
          isLoading={false}
        />
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})