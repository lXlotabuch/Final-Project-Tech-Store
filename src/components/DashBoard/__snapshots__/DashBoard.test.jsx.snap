// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DashBoard component shaphot test 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ dydVzA"
  >
    <aside
      class="sc-hKgILt emLhCT"
    >
      <p>
        My Dashboard
      </p>
    </aside>
    <div
      class="sc-dlfnbm cwrOeu"
    >
      <h3>
        Account Information
      </h3>
    </div>
    <div
      class="sc-eCssSg bJHaJZ"
    >
      <div
        class="sc-jSgupP ddDNyR"
      >
        <div
          class="sc-gKsewC jXkQxd"
        >
          <h4>
            Contact Information
          </h4>
          <div>
            <div
              class="sc-iBPRYJ gLlUwN"
            >
              <p>
                <span
                  class="sc-jrAGrp jCuYjM"
                >
                  First Name:
                </span>
                 
              </p>
              <p>
                <span
                  class="sc-jrAGrp jCuYjM"
                >
                  Last Name:
                </span>
                 
              </p>
              <p>
                <span
                  class="sc-jrAGrp jCuYjM"
                >
                  Phone:
                </span>
                 
              </p>
            </div>
            <div>
              <button
                data-testid="setInfo"
                id="setInfo"
                type="submit"
              >
                Edit
              </button>
              <button
                id="setPassword"
                type="submit"
              >
                Change Password
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jSgupP ddDNyR"
      >
        <div
          class="sc-gKsewC jXkQxd"
        >
          <h4>
            Newsletters
          </h4>
          <div>
            <div
              class="sc-iBPRYJ gLlUwN"
            >
              <div
                class="sc-fubCfw hYEFQe"
              >
                <p>
                  You don't subscribe to our newsletter.
                </p>
                <button
                  type="submit"
                >
                  Subscribe
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-jSgupP ddDNyR"
      >
        <div
          style="margin-top: 20px;"
        >
          <div>
            <h5
              style="text-align: center; font-size: 20px; padding-bottom: 20px;"
            >
              My Orders
            </h5>
          </div>
          <div
            class="sc-hBEYos iwRQNr"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
